### DISCUSSION TITLE
What are `badCodes`?

### DISCUSSION THREAD

**Discussion by ghost** [opendcs/opendcs#817] [Q&A] [answered]
Where can I read more about these bad codes? Is the idea that the presence of these characters in a pseudo-binary message (or any other message format) indicates the absence of data or error in the data?

[	public static final String badCodes = "?MTUBIQW";](https://github.com/opendcs/opendcs/blob/edfee8aece99414dc939a277ce1279f97c89a5e4/java/opendcs/src/main/java/lrgs/common/DcpMsg.java#L906)

**Comment by MikeNeilson** [comment]
There's definitely some documentations on the GOES DCS system, but I'm not finding them quickly in the location I expected. Digging through this list may have something: https://www.noaasis.noaa.gov/GOES/GOES_DCS/history.html

@NathanHolcomb ?

**Comment by ghost** [comment, marked as answer]
Should be this - https://opendcs-env.readthedocs.io/en/stable/legacy-lrgs-userguide.html#failure-code
